subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java

```diff
@@ -491,6 +491,14 @@ private void writePlugin(String alias, String id, @Nullable String context, bool
         writeLn();
     }
 
+    /**
+     * Java compiler would fail to compile sources that have illegal unicode escape characters in the comments.
+     * Such characters could be accidentally introduced by an unescaped backslash.
+     */
+    private static String escapeBackslashForJavaComment(String s) {
+        return s.replace("\\", "\\\\");
```

Comment:
I am not too fond of this solution, the end result in the javadoc is something like:
`C:\\Users\\user\\Documents\\ultimate plugin` where all `\` a doubled.

Why don't we address the exact problem and replace `\u` with `\u005cu` which both an IDE (when viewing Javadoc) and the generated Javadoc would render as `\u`.

Reply:
This is a nice solution, although at the same time if the javadoc is not rendered, then the path looks even more unreadble/broken: `C:\Users\u005cuser\Documents\u005cultimate plugin`

Reply:
It was decided to go with @ljacomet 's solution 👍